src/webview/js/js.js

```diff
@@ -637,16 +628,30 @@ const handleLongPress = (target: Element) => {
   // a library we can use which strives to handle all that complexity, or
   // get long-press events from the platform.
   if (!lastTouchEventTimestamp || Date.now() - lastTouchEventTimestamp < 500) {
-    return;
-  }
+    if (target.matches('a')) {
```

Comment:
You already gave some explanation that there is weirdness in how long-press works with links.
I am not sure if this solution is the right one (I really don't know :) )
It will be better to extract this behavior in a separate commit.

The behavior of long press on links is likely to differ between Android and iOS.
Make sure to test on both platforms and to mention that in the commit message.

Reply:
Long-press works weirdly with mobile devices, since `MouseEvents` sort of conflict with `TouchEvents`. This solution seems to be the only one available since there are various issues introduced if the longPress events for URLs and the messages/headers are decoupled (the action sheet launches and the URL is opened, instead of one of them happening in each case). I could possibly use a library which exposes longPress properly in js (as mentioned in the comments preceding this code), but all of the libraries I found either didn't work properly (not as desired for this use-case) or was poorly maintained.

Plus, with this change (coupled with the previous comment about changing `MessageListEventLongPress` to include another target for `link`) the code becomes much less cluttered, since all the three targets are confined to one scope.

I will check out long presses on Android and update the commit message (tested on iOS for now). Thanks!

Reply:
This bit of logic doesn't belong here inside this conditional. The reason is that it makes the code much more complex to understand.

One way to see that: try to describe what the meaning of this code is (this `if (target.matches('a')) ...` block, when it's located here), and to explain why it works correctly. I think you'll find that is pretty complicated to do successfully -- and moreover that it contradicts what's there in the text, in particular the name `handleLongPress`.

---

I agree with Boris's advice to separate out this change in its own commit. I don't yet understand what your purpose is in moving this bit of code here from the `click` handler, so it requires some explanation... and it will be a lot easier to successfully explain it if it's in its own commit.

(Like I said above, it still won't belong quite *here*, but once we understand what this change is intended to do we'll be better able to advise on how to adjust it.)

Reply:
I see the point that it's hard to understand, but the logic gets pretty messed up if we try and un-bind it.

An issue arises where both short press and long press can be triggered from clicking the message.

To remedy the understandability of the code, I have made a few changes within the code.

- Changed the method name from `handleLongPress` to `handleClicksOnMessage` since that is basically what the method is doing, it handles behaviour of a users interaction with the body of the message.
- Added a comment detailing what the problem is, and how the implemented logic works.

> // This method is a little convoluted, since it aims to provide a solution
> // to the long press event in-availability. If this solution is not
> // provided, both the short presses and the long presses can trigger at the
> // same time when a user tries to perform an action on a message.
> //
> // The first if checks if the person is clicking the link, and if
> // the holding time is less than 500ms.
> // If both of the conditions check, and the target of the click is a URL,
> // this script returns a `sendMessage` with type `url`, else, script
> // returns with `sendMessage` type `longPress`, and an href if the element
> // the `longPress` is performed on is a link.

That said I did move the change into it's own commit.

Reply:
Said issue can be experienced like [this](https://imgur.com/phH6B5C).